catalog/admin/module/catalog-admin-module-maplayers/src/main/webapp/map-layers/index.js

```diff
@@ -0,0 +1,299 @@
+import React from 'react'
+
+import { Map } from 'immutable'
+import { connect } from 'react-redux'
+
+import CircularProgress from 'material-ui/CircularProgress'
+import ContentAdd from 'material-ui/svg-icons/content/add'
+import DeleteIcon from 'material-ui/svg-icons/action/delete'
+import FlatButton from 'material-ui/FlatButton'
+import Flexbox from 'flexbox-react'
+import FloatingActionButton from 'material-ui/FloatingActionButton'
+import IconButton from 'material-ui/IconButton'
+import MenuItem from 'material-ui/MenuItem'
+import Paper from 'material-ui/Paper'
+import RaisedButton from 'material-ui/RaisedButton'
+import SelectField from 'material-ui/SelectField'
+import Snackbar from 'material-ui/Snackbar'
+import TextField from 'material-ui/TextField'
+import muiThemeable from 'material-ui/styles/muiThemeable'
+import { List, ListItem } from 'material-ui/List'
+
+import Mount from 'react-mount'
+
+import AceEditor from 'react-ace'
+import 'brace/mode/json'
+import 'brace/theme/github'
+
+import options from './options'
+import {
+  // actions
+  fetch,
+  update,
+  save,
+  validate,
+  reset,
+  message,
+
+  // selectors
+  getProviders,
+  isLoading,
+  hasChanges,
+  getMessage
+} from './reducer'
+
+const submittingStyle = {
+  position: 'absolute',
+  top: 0,
+  bottom: 0,
+  right: 0,
+  left: 0,
+  background: 'rgba(0, 0, 0, 0.1)',
+  zIndex: 9001,
+  display: 'flex'
+}
+
+const Spinner = ({ submitting = false, children }) => (
```

Comment:
❓ Why did you decide to put all of the components of this app in this one file instead of splitting it up?